### PR TITLE
 feat/season-plugin-option

### LINKED ISSUE #2: Add season plugin option
At the moment it's always on.

It should be able to set and forget, having it stop/start for each holiday season...

### PR DESCRIPTION
**Pull request by PaulieScanlon**
- add season `start` and `end`

The only slight snag with this is: 

When a user sets the dates via the plugin options it **must** be a valid date object which includes the year. Naturally, next year this wouldn't start the snow unless the user changes the year in the options

```javascript
    {
      resolve: "@raae/gatsby-plugin-let-it-snow",
      options: {
        season: {
          start: new Date("December 1, 2021"), // date including year
          end: new Date("January 1, 2022"), // date including year
        },
      },
    },
```

### CODE REVIEW

File: plugin/gatsby-browser.js [hunk 1]
```diff
@@ -40,5 +41,14 @@ export const onInitialClientRender = (_, options) => {
     }
   };
 
-  frame();
+  if (season) {
```
Comment (PaulieScanlon) [inline plugin/gatsby-browser.js]: We could do `if(season && new Date() ...)` not sure if you're a fan of nested `if` statements or not?

**Comment by raae** [comment]
I think you can do custom validation, ie. letting folks write: 

    {
      resolve: "@raae/gatsby-plugin-let-it-snow",
      options: {
        season: {
          start: 12-01
          end: 01-01
        },
      },
    },

And then custom verify that the string prefixed with the year is date...

I also think these dates should be the defaults if no season is set (making it a breaking change 🤯).

**Comment by PaulieScanlon** [comment]
Dates are hard but to me 12-01 is the 12th of January so maybe the long hand way will work better so folks have to write it out saving confusion.

Also agree it should have a default.

Breaking change is ok. It can be the first stable version 1.0.0

... but we should probably add a CHANGELOG.md so it's clear, as well as of course, updating the README.md

**Comment by raae** [comment]
What about your proposal + a flag for repeat every year? That is `true` by default?

**Comment by raae** [comment]
See #12 for changelog

**Comment by PaulieScanlon** [comment]
I'll have to look into it but you need the year in the date sting for it to be valid, and then to be able to run `getTime` on the date object which was the easiest way I could think of to handle the condition. 

I'd be inclined to leave it and if folks want it next year they can update the date string.

**Comment by PaulieScanlon** [comment]
I think as well using the date string folks could include a time

E.g 

`end: new Date("December 31, 2021 11:59:00")`

And the `getTime` condition would still work

File: plugin/gatsby-browser.js [hunk 2]
```diff
@@ -40,5 +41,14 @@ export const onInitialClientRender = (_, options) => {
...
+    if (
+      new Date(now).getTime() >= new Date(season.start).getTime() &&
```
Comment (raae) [inline plugin/gatsby-browser.js]: **question:** Aren't season.start already a date?

**Comment by raae** [comment]
I really want it to start again next year, might look into it a little.

File: plugin/gatsby-browser.js (hunk 2 above)
Comment (PaulieScanlon) [reply plugin/gatsby-browser.js]: They are but they get passed as `string` so you can't run `getTime()` on the unless the `new Date` constructor is used

**Comment by PaulieScanlon** [comment]
There's other ways to do this by changing the config to use a `start` and `end` `month` / `day`

Maybe something like this?


```javascript
  plugins: [
    {
      resolve: "@raae/gatsby-plugin-let-it-snow",
      options: {
        season: {
          start: {
            day: 1,
            month: 12,
          },
          end: {
            day: 31,
            month: 12,
          },
        },
      },
    },
  ],
```

Then i guess those values could be used to construct a `new Date` and the `year` bit gets added by the plugin. This would then repeat each year and because a date is still constructed, the `getTime()` method to evaluate if it's within range would still work.


If you like the above API i can make the change, it's no trouble at all Queen!

File: plugin/gatsby-browser.js (hunk 2 above)
Comment (raae) [reply plugin/gatsby-browser.js]: I see!

**Comment by raae** [comment]
Let's ask Twitter 🎉

**Comment by raae** [comment]
Let's do it as you proposed and I will try to sneak in a repeat flag at some point.

File: plugin/gatsby-browser.js (hunk 1 above)
Comment (raae) [reply plugin/gatsby-browser.js]: No real preference.

File: plugin/gatsby-node.js [hunk 3]
```diff
@@ -4,5 +4,11 @@ exports.pluginOptionsSchema = ({ Joi }) => {
     colors: Joi.array()
       .items(Joi.string().default("#ffffff"))
       .description("Array of hex color values"),
+    season: Joi.object()
+      .keys({
+        start: Joi.date(),
```
Comment (raae) [inline plugin/gatsby-node.js]: **nitpick:** Add default dates.

**Review by raae — CHANGES_REQUESTED** [review]
I can add the defaults, if you add me to your fork.

**Review by PaulieScanlon — COMMENTED** [review]
Lovely stuff!

**Pull request merged**